core/domain/user_jobs_one_off.py

```diff
@@ -68,6 +68,29 @@ def reduce(key, version_and_exp_ids):
                     edited_exploration_ids))
 
 
+class UsernameLengthDistributionOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job for calculating the distribution of username lengths."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(item):
+        if item.username is not None:
+            yield (len(item.username), {
```

Comment:
If you just need the number of usernames, you can just yield (len(item.username), 1).